library/bignum.c

```diff
@@ -131,15 +131,17 @@ int mbedtls_mpi_safe_cond_assign(mbedtls_mpi *X,
 
     MBEDTLS_MPI_CHK(mbedtls_mpi_grow(X, Y->n));
 
-    mbedtls_ct_condition_t do_assign = mbedtls_ct_bool(assign);
+    {
```

Comment:
(comment) we normally fix these kinds of warnings by hoisting the variable declarations up to the top and initialising. Here, both `assign` and `do_not_assign` could be declared and initialised correctly before the `CHK` (`MPI_VALIDATE_RET()` is a no-op these days)

Reply:
I didn't do this because it increases code size slightly.

Reply:
It is very minor though, it might be better to prioritise tidier code. Up to you.

Reply:
I have a mild preference for consistency of pattern, since it makes it easier for people writing code in the future to know what to do. However, code size saving 65265 vs 65249 is enough to keep this - and suggests we should revisit other functions!